parquet/src/test/java/org/apache/iceberg/parquet/TestParquet.java

```diff
@@ -219,6 +228,191 @@ public void testTwoLevelList() throws IOException {
     assertThat(recordRead.get("topbytes")).isEqualTo(expectedBinary);
   }
 
+  @Test
+  public void testStreamClosedProperly() throws IOException {
+    // test for input
+    {
+      class TestStreamClosedProperlyStream extends SeekableInputStream
+          implements DelegatingInputStream {
+        boolean thisClosed = false;
+        boolean delegateClosed = false;
+
+        {
```

Comment:
it feels weird that reset needs to be called here. I believe this can be removed